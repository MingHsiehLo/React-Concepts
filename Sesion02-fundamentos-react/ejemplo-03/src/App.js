import React from 'react';

class App extends React.Component {
  state = {
    currentCount: 0
  }

  onAddition = () => this.setState({ currentCount: this.state.currentCount + 1 })
  onSubstraction = () => this.setState({ currentCount: this.state.currentCount - 1 })

  render() {
    return (
      <div className="App">
        <h1>Virtual DOM</h1>
        <h2>{this.state.currentCount}</h2> 
        <button onClick={() => this.onSubstraction()}>-</button>
        <button onClick={() => this.onAddition()}>+</button>
      </div>
    );
  }
}

export default App;
