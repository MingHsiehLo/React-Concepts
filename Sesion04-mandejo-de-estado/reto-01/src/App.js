import React from 'react'

class App extends React.Component {
  state = {
    data: []
  }

  render() {
    return (
      <div>Aquí irá la información a desplegar</div>
    )
  }
}

export default App;
