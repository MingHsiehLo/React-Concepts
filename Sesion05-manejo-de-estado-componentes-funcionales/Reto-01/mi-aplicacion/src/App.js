import React from 'react'

function App(props) {
  const [message, setMessage] = React.useState('')

  return (
    <>
      <input
        value={message}
        onChange={e => setMessage(e.target.value)}
        placeholder="Escribe aquÃ­..."
      />
      <p>{ message }</p>
    </>
  )
}

export default App;
