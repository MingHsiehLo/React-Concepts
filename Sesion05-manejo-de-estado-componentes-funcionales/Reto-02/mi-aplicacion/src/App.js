import React from 'react'

function App(props) {
  const [message, setMessage] = React.useState('')

  React.useEffect(() => {
    setMessage('Cámbiame...')
  }, [])

  return (
    <>
      <input
        value={message}
        onChange={e => setMessage(e.target.value)}
        placeholder="Escribe aquí..."
      />
      <p>{ message }</p>
    </>
  )
}

export default App;
