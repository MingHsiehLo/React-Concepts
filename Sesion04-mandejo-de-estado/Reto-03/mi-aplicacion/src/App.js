import React from 'react'

import './App.css'
import Character from './Character'

class App extends React.Component {
  state = {
    data: []
  }

  componentDidMount() {
    fetch('https://rickandmortyapi.com/api/character/')
    .then(r => r.json())
    .then(r => this.setState((state, props) => ({ data: r.results })))
  }

  render() {
    return (
      <div className="App">
        { this.state.data.map(d => {
          return (
            <Character
              key={d.id}
              image={d.image}
              name={d.name}
              specie={d.species}
            />
          )
        })}
      </div>
    )
  }
}

export default App;
