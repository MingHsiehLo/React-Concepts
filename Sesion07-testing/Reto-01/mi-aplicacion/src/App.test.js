import React from 'react'

import App from './App'

test('Render without crash', () => {
  const component = shallow(<App />)
  expect(component).toMatchSnapshot()
})

test('Change input changes p', () => {
  const component = mount(<App />)

  // Testing initial state
  expect(component.find('p').equals(<p>Cámbiame...</p>)).toEqual(true)
  // Testing a change in the input
  component.find('input').simulate('change', {target:{value: "Hola Mundo"}})
  expect(component.find('p').equals(<p>Cámbiame...</p>)).toEqual(false)
  expect(component.find('p').equals(<p>Hola Mundo</p>)).toEqual(true)
})
