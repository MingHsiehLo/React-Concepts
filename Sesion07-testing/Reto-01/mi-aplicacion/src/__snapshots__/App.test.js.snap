// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without crash 1`] = `ShallowWrapper {}`;
